### ISSUE TITLE
Array destructuring each triggers false match Style/HashEachMethods

### ISSUE THREAD

**Issue by jcoyne** [rubocop/rubocop#12507]
In rubocop 1.58.0 this code triggers Style/HashEachMethods.
```
arry = [[1, 2], [3, 4]
arry.each { |_x, y| puts y }
```

If you correct it you get the error:
```
undefined method `each_value' for [[1, 2], [3, 4]]:Array (NoMethodError)
```

Clearly de-structuring in a block is not sufficient evidence to determine the callee is a Hash.

**Comment by koic** [comment]
This issue is a dup with https://github.com/rubocop/rubocop/issues/12444.

**Comment by koic** [comment]
And, RuboCop cannot determine the type of the receiver, this cop has already been marked as unsafe:
https://docs.rubocop.org/rubocop/1.58/cops_style.html#stylehasheachmethods

Moreoverer, you can address this issue with the following configuration:

```yaml
# .rubocop.yml
Style/HashEachMethods:
  AllowedReceivers: ['arry']
```
